Test fail on base dir due to parameter changes on `findFiles`,

```
dev-shulhan ms 2 % go test -v ./...
# github.com/shuLhan/go-bindata
./convert_test.go:24:18: not enough arguments in call to findFiles
        have (string, *regexp.Regexp, bool, *[]Asset, []*regexp.Regexp, map[string]int, map[string]bool)
        want (string, *regexp.Regexp, bool, *[]Asset, []*regexp.Regexp, []*regexp.Regexp, map[string]int, map[string]bool)
./convert_test.go:41:18: not enough arguments in call to findFiles
        have (string, *regexp.Regexp, bool, *[]Asset, []*regexp.Regexp, map[string]int, map[string]bool)
        want (string, *regexp.Regexp, bool, *[]Asset, []*regexp.Regexp, []*regexp.Regexp, map[string]int, map[string]bool)
./convert_test.go:50:17: not enough arguments in call to findFiles
        have (string, *regexp.Regexp, bool, *[]Asset, []*regexp.Regexp, map[string]int, map[string]bool)
        want (string, *regexp.Regexp, bool, *[]Asset, []*regexp.Regexp, []*regexp.Regexp, map[string]int, map[string]bool)
./convert_test.go:75:18: not enough arguments in call to findFiles
        have (string, *regexp.Regexp, bool, *[]Asset, []*regexp.Regexp, map[string]int, map[string]bool)
        want (string, *regexp.Regexp, bool, *[]Asset, []*regexp.Regexp, []*regexp.Regexp, map[string]int, map[string]bool)
./convert_test.go:92:18: not enough arguments in call to findFiles
        have (string, *regexp.Regexp, bool, *[]Asset, []*regexp.Regexp, map[string]int, map[string]bool)
        want (string, *regexp.Regexp, bool, *[]Asset, []*regexp.Regexp, []*regexp.Regexp, map[string]int, map[string]bool)
FAIL    github.com/shuLhan/go-bindata [build failed]
```

--- a/config.go
+++ b/config.go
@@ -139,7 +139,8 @@ type Config struct {
 	// will match any .gitignore file.
 	//
 	// This parameter can be provided multiple times.
-	Ignore []*regexp.Regexp
+	Ignore  []*regexp.Regexp
+	Include []*regexp.Regexp
 
 	// MD5Checksum is a flag that, when set to true, indicates to calculate
 	// MD5 checksums for files.
@@ -155,6 +156,7 @@ func NewConfig() *Config {
 	c.Debug = false
 	c.Output = "./bindata.go"
 	c.Ignore = make([]*regexp.Regexp, 0)
+	c.Include = make([]*regexp.Regexp, 0)
 	return c
 }
 

--- a/convert.go
+++ b/convert.go
@@ -31,7 +31,7 @@ func Translate(c *Config) error {
 	var visitedPaths = make(map[string]bool)
 	// Locate all the assets.
 	for _, input := range c.Input {
-		err = findFiles(input.Path, c.Prefix, input.Recursive, &toc, c.Ignore, knownFuncs, visitedPaths)
+		err = findFiles(input.Path, c.Prefix, input.Recursive, &toc, c.Ignore, c.Include, knownFuncs, visitedPaths)
 		if err != nil {
 			return err
 		}
@@ -125,6 +125,7 @@ func findFiles(
 	recursive bool,
 	toc *[]Asset,
 	ignore []*regexp.Regexp,
+	include []*regexp.Regexp,
 	knownFuncs map[string]int,
 	visitedPaths map[string]bool,
 ) (err error) {
@@ -185,6 +186,15 @@ func findFiles(
 				break
 			}
 		}
+
+		for _, re := range include {
+			if re.MatchString(asset.Path) {
+				ignoring = false
+				break
+			}
+			ignoring = true
+		}
+
 		if ignoring {
 			continue
 		}
@@ -193,7 +203,7 @@ func findFiles(
 			if recursive {
 				recursivePath := filepath.Join(dir, file.Name())
 				visitedPaths[asset.Path] = true
-				findFiles(recursivePath, prefix, recursive, toc, ignore, knownFuncs, visitedPaths)
+				findFiles(recursivePath, prefix, recursive, toc, ignore, include, knownFuncs, visitedPaths)
 			}
 			continue
 		}
@@ -212,7 +222,7 @@ func findFiles(
 
 			if _, ok := visitedPaths[linkPath]; !ok {
 				visitedPaths[linkPath] = true
-				findFiles(asset.Path, prefix, recursive, toc, ignore, knownFuncs, visitedPaths)
+				findFiles(asset.Path, prefix, recursive, toc, ignore, include, knownFuncs, visitedPaths)
 			}
 			continue
 		}

--- a/convert_test.go
+++ b/convert_test.go
@@ -21,7 +21,7 @@ func TestFindFiles(t *testing.T) {
 	var visitedPaths = make(map[string]bool)
 	prefix := regexp.MustCompile("testdata/dupname")
 
-	err := findFiles("testdata/dupname", prefix, true, &toc, []*regexp.Regexp{}, knownFuncs, visitedPaths)
+	err := findFiles("testdata/dupname", prefix, true, &toc, []*regexp.Regexp{}, []*regexp.Regexp{}, knownFuncs, visitedPaths)
 	if err != nil {
 		t.Errorf("expected to be no error: %+v", err)
 	}
@@ -38,7 +38,7 @@ func TestFindFilesWithSymlinks(t *testing.T) {
 	var visitedPaths = make(map[string]bool)
 	prefix := regexp.MustCompile("testdata/symlinkSrc")
 
-	err := findFiles("testdata/symlinkSrc", prefix, true, &tocSrc, []*regexp.Regexp{}, knownFuncs, visitedPaths)
+	err := findFiles("testdata/symlinkSrc", prefix, true, &tocSrc, []*regexp.Regexp{}, []*regexp.Regexp{}, knownFuncs, visitedPaths)
 	if err != nil {
 		t.Errorf("expected to be no error: %+v", err)
 	}
@@ -47,7 +47,7 @@ func TestFindFilesWithSymlinks(t *testing.T) {
 	visitedPaths = make(map[string]bool)
 	prefix = regexp.MustCompile("testdata/symlinkParent")
 
-	err = findFiles("testdata/symlinkParent", prefix, true, &tocTarget, []*regexp.Regexp{}, knownFuncs, visitedPaths)
+	err = findFiles("testdata/symlinkParent", prefix, true, &tocTarget, []*regexp.Regexp{}, []*regexp.Regexp{}, knownFuncs, visitedPaths)
 	if err != nil {
 		t.Errorf("expected to be no error: %+v", err)
 	}
@@ -72,7 +72,7 @@ func TestFindFilesWithRecursiveSymlinks(t *testing.T) {
 	var visitedPaths = make(map[string]bool)
 	prefix := regexp.MustCompile("testdata/symlinkRecursiveParent")
 
-	err := findFiles("testdata/symlinkRecursiveParent", prefix, true, &toc, []*regexp.Regexp{}, knownFuncs, visitedPaths)
+	err := findFiles("testdata/symlinkRecursiveParent", prefix, true, &toc, []*regexp.Regexp{}, []*regexp.Regexp{}, knownFuncs, visitedPaths)
 	if err != nil {
 		t.Errorf("expected to be no error: %+v", err)
 	}
@@ -89,7 +89,7 @@ func TestFindFilesWithSymlinkedFile(t *testing.T) {
 	var visitedPaths = make(map[string]bool)
 	prefix := regexp.MustCompile("testdata/symlinkFile")
 
-	err := findFiles("testdata/symlinkFile", prefix, true, &toc, []*regexp.Regexp{}, knownFuncs, visitedPaths)
+	err := findFiles("testdata/symlinkFile", prefix, true, &toc, []*regexp.Regexp{}, []*regexp.Regexp{}, knownFuncs, visitedPaths)
 	if err != nil {
 		t.Errorf("expected to be no error: %+v", err)
 	}

--- a/go-bindata/main.go
+++ b/go-bindata/main.go
@@ -39,14 +39,16 @@ var (
 
 // List of error messages.
 var (
-	ErrInvalidIgnoreRegex = errors.New("Invalid -ignore regex pattern")
-	ErrInvalidPrefixRegex = errors.New("Invalid -prefix regex pattern")
-	ErrNoInput            = errors.New("Missing <input directories>")
+	ErrInvalidIgnoreRegex  = errors.New("Invalid -ignore regex pattern")
+	ErrInvalidIncludeRegex = errors.New("Invalid -include regex pattern")
+	ErrInvalidPrefixRegex  = errors.New("Invalid -prefix regex pattern")
+	ErrNoInput             = errors.New("Missing <input directories>")
 )
 
 // List of local variables.
 var (
 	argIgnore  []string
+	argInclude []string
 	argVersion bool
 	argPrefix  string
 	cfg        *bindata.Config
@@ -114,6 +116,7 @@ func initArgs() {
 	flag.StringVar(&cfg.Tags, "tags", cfg.Tags, "Optional set of build tags to include.")
 	flag.UintVar(&cfg.Mode, "mode", cfg.Mode, "Optional file mode override for all files.")
 	flag.Var((*AppendSliceValue)(&argIgnore), "ignore", "Regex pattern to ignore")
+	flag.Var((*AppendSliceValue)(&argInclude), "include", "Regex pattern to include")
 }
 
 //
@@ -147,6 +150,11 @@ func parseArgs() (err error) {
 	}
 
 	err = parseIgnore()
+	if err != nil {
+		return
+	}
+
+	err = parseInclude()
 	if err != nil {
 		return
 	}
@@ -185,6 +193,21 @@ func parseIgnore() (err error) {
 		}
 
 		cfg.Ignore = append(cfg.Ignore, ignoreVal)
+	}
+
+	return
+}
+
+func parseInclude() (err error) {
+	var includeVal *regexp.Regexp
+
+	for _, pattern := range argInclude {
+		includeVal, err = regexp.Compile(pattern)
+		if err != nil {
+			return ErrInvalidIncludeRegex
+		}
+
+		cfg.Include = append(cfg.Include, includeVal)
 	}
 
 	return

--- a/go-bindata/main_test.go
+++ b/go-bindata/main_test.go
@@ -93,7 +93,8 @@ func TestParseArgs(t *testing.T) {
 			Input: []bindata.InputConfig{{
 				Path: argInputPath,
 			}},
-			Ignore: defConfig.Ignore,
+			Ignore:  defConfig.Ignore,
+			Include: defConfig.Include,
 		},
 	}, {
 		desc: `With "-pkg ` + argPkg + `"`,
@@ -108,7 +109,8 @@ func TestParseArgs(t *testing.T) {
 			Input: []bindata.InputConfig{{
 				Path: argInputPath,
 			}},
-			Ignore: defConfig.Ignore,
+			Ignore:  defConfig.Ignore,
+			Include: defConfig.Include,
 		},
 	}, {
 		desc: `With "-o ` + argOutFile + `" (package name should be "` + argOutPkg + `")`,
@@ -123,7 +125,8 @@ func TestParseArgs(t *testing.T) {
 			Input: []bindata.InputConfig{{
 				Path: argInputPath,
 			}},
-			Ignore: defConfig.Ignore,
+			Ignore:  defConfig.Ignore,
+			Include: defConfig.Include,
 		},
 	}, {
 
@@ -140,7 +143,8 @@ func TestParseArgs(t *testing.T) {
 			Input: []bindata.InputConfig{{
 				Path: argInputPath,
 			}},
-			Ignore: defConfig.Ignore,
+			Ignore:  defConfig.Ignore,
+			Include: defConfig.Include,
 		},
 	}}
 

